### PR TITLE
[MM-17200] Add render methods for new guest messages

### PR DESCRIPTION
**Pull request by mgdelacroix**
#### Ticket Link
https://mattermost.atlassian.net/browse/MM-17200

#### Related Pull Requests
- [Has server changes](https://github.com/mattermost/mattermost-server/pull/11803)
- [Has redux changes](https://github.com/mattermost/mattermost-redux/pull/893)

### CODE REVIEW

**Comment by mgdelacroix** [comment]
The ticket only refers to the system post that appears when a guest is added to a channel by another user, but these PRs add a new message to the system post that appears when a guest joins a channel (for example, through an invitation).

@wiersgallak can you please confirm this second change?

**Comment by mgdelacroix** [comment]
Checked that the test failure is caused by `mattermost-redux` not pointing to the right commit, should be fixed when the redux PR is merged.

File: components/post_markdown/system_message_helpers.jsx [hunk 1]
```diff
@@ -44,6 +44,18 @@ function renderJoinChannelMessage(post) {
     );
 }
 
+function renderGuestJoinChannelMessage(post) {
+    const username = renderUsername(post.props.username);
+
+    return (
+        <FormattedMessage
+            id='api.channel.guest_join_channel.post_and_forget'
+            defaultMessage='{username} joined the channel as guets.'
```
Comment (wiersgallak) [inline components/post_markdown/system_message_helpers.jsx]: Please update to: "{username} joined the channel as a guest."  (note spelling on guest and adding a "a")

File: components/post_markdown/system_message_helpers.jsx [hunk 2]
```diff
@@ -72,6 +84,22 @@ function renderAddToChannelMessage(post) {
...
 
+function renderAddGuestToChannelMessage(post) {
+    const username = renderUsername(post.props.username);
+    const addedUsername = renderUsername(post.props.addedUsername);
+
+    return (
+        <FormattedMessage
+            id='api.channel.add_guest.added'
+            defaultMessage='{addedUsername} added to the channel as guest by {username}.'
+            values={{
```
Comment (wiersgallak) [inline components/post_markdown/system_message_helpers.jsx]: Please update to : "{addedUsername} added to the channel as a guest by {username}."

File: i18n/en.json [hunk 3]
```diff
@@ -1642,8 +1642,10 @@
   "announcement_bar.error.site_url_gitlab.full": "Please configure your [Site URL](https://docs.mattermost.com/administration/config-settings.html#site-url) in the [System Console](/admin_console/environment/web_server) or in gitlab.rb if you're using GitLab Mattermost.",
   "announcement_bar.error.site_url.full": "Please configure your [Site URL](https://docs.mattermost.com/administration/config-settings.html#site-url) in the [System Console](/admin_console/environment/web_server).",
   "announcement_bar.notification.email_verified": "Email verified",
+  "api.channel.add_guest.added": "{addedUsername} added to the channel as guest by {username}.",
```
Comment (wiersgallak) [inline i18n/en.json]: Please update to: "{addedUsername} added to the channel as a guest by {username}."

File: i18n/en.json [hunk 4]
```diff
@@ -1642,8 +1642,10 @@
...
   "api.channel.add_member.added": "{addedUsername} added to the channel by {username}.",
   "api.channel.delete_channel.archived": "{username} archived the channel.",
+  "api.channel.guest_join_channel.post_and_forget": "{username} joined the channel as guest.",
```
Comment (wiersgallak) [inline i18n/en.json]: Please update to: "{username} joined the channel as a guest."

**Comment by mgdelacroix** [comment]
Thanks @wiersgallak! Updated

**Comment by lindalumitchell** [comment]
Re-assigned QA review to Rohitesh, as he and Ogi are QA point people on guest accounts.

**Comment by lindalumitchell** [comment]
(Sorry; closed with a bad tab/Enter combo and re-opened. :) )

**Pull request merged**